zerver/views/registration.py

```diff
@@ -324,6 +333,38 @@ def registration_helper(
                     "key": key,
                 },
             )
+
+        elif prereg_realm.data_import_metadata.get("import_from") == "slack":
+            context: dict[str, Any] = {
+                "key": key,
+            }
+
+            saved_slack_access_token = prereg_realm.data_import_metadata.get("slack_access_token")
+            if saved_slack_access_token or slack_access_token:
+                if slack_access_token and slack_access_token != saved_slack_access_token:
+                    # Verify slack token access.
+                    from zerver.data_import.slack import (
+                        SLACK_IMPORT_TOKEN_SCOPES,
+                        check_token_access,
+                    )
+
+                    check_token_access(slack_access_token, SLACK_IMPORT_TOKEN_SCOPES)
```

Comment:
This raises exceptions if you don't have the right permissions set, which means currently you'd get a 500

Reply:
Ah! Nice catch. Changed to raise an exception and show the error.